bundles/org.openhab.binding.iotawatt/src/main/java/org/openhab/binding/iotawatt/internal/client/IoTaWattClient.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.iotawatt.internal.client;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.http.HttpMethod;
+import org.openhab.binding.iotawatt.internal.model.StatusResponse;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * Encapsulates the communication with the IoTaWatt device.
+ *
+ * @author Peter Rosenberg - Initial contribution
+ */
+@NonNullByDefault
+public class IoTaWattClient {
+    private static final String REQUEST_URL = "http://%s/status?state=&inputs=&outputs=";
+
+    private final Logger logger = LoggerFactory.getLogger(IoTaWattClient.class);
+
+    public final String hostname;
+    private final HttpClient httpClient;
+    private final Gson gson;
+
+    public IoTaWattClient(String hostname, HttpClient httpClient, Gson gson) {
+        this.httpClient = httpClient;
+        this.hostname = hostname;
+        this.gson = gson;
+    }
+
+    /**
+     * Fetch the current status from the device.
+     * The errors are handled by the caller to update the Thing status accordingly.
+     */
+    public Optional<StatusResponse> fetchStatus()
+            throws ExecutionException, InterruptedException, TimeoutException, URISyntaxException {
+        final URI uri = new URI(String.format(REQUEST_URL, hostname));
+        final Request request = httpClient.newRequest(uri).method(HttpMethod.GET);
+        final ContentResponse response = request.send();
+        final String content = response.getContentAsString();
+        @Nullable
+        final StatusResponse statusResponse = gson.fromJson(content, StatusResponse.class);
+        logger.trace("statusResponse: {}", statusResponse);
+        if (statusResponse.inputs() == null) {
+            logger.error("List of inputs in response from IoTaWatt is null on device {}.", hostname);
```

Comment:
See description for when to use error and warn:
https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
Yes, I'm aware of these rules and feel this one under `error` applies:
`in case some code fails irrecoverably and the user should report it as a severe bug.`
Do you think it would be better as `warn`?

Reply:
Yes, warn would be best.